docs/failsafe-controller.md

```diff
@@ -0,0 +1,192 @@
+```
+
+In this example, the first line will create a `net.jodah.failsafe.SyncFailsafe` object. The second line will create a `net.jodah.failsafe.AsyncFailsafe` object from it and the next 4 lines registers event listeners. Additional lines could be added to further configure Failsafe if required. The final line is the one that actually triggers the start of an execution which returns a `java.util.concurrent.Future` allowing access to the result at the end and also provides the ability to cancel the execution.
+
+The Failsafe controller is not a replacement for Failsafe but does require replacing the first line above. One way to achieve this is by defining a variable in your class that can be injected through the constructor at build time. This variable could be defined as a function like this:
+```
+  private final Function<RetryPolicy, SyncFailsafe<MyObject>> createFailsafeCreator;
+```
+With a package private constructor like this:
+```
+  MyClass(Function<RetryPolicy, SyncFailsafe<MyObject>> createFailsafeCreator) {
+    this.createFailsafeCreator = createFailsafeCreator;
+  }
+```
+The normal constructor could simply call that constructor like this:
+```
+ this(Failsafe::with)
+```
+From this point the example above would be re-written as:
+```
+  Future<MyObject> future = createFailsafeCreator.apply(myRetryPolicy)
+    .with(myScheduledExecutor)
+    .onRetry(this::logFailure)
+    .onAbort(this::logInterruptionAndRecreate)
+    .onFailure(this::logAndRecreateIfNotCancelled)
+    .onSuccess(this::logAndSetCreated)
+    .get(this::create);
+```
+While testing, the constructor would be called as follow:
+```
+  FailsafeController controller = new FailsafeController("some name");
+
+  MyClass mine = new MyClass(controller::with);
+```
+From this point on, the controller instance becomes your point of contact to control, mock, or script how Failsafe will respond to each executions and attempts.
+
+### Actions
+Each time Failsafe makes an attempt, the controller intercepts the call to the task specified (see 7th line in the above example) and figures out what to do. This could be:
+
+* return something
+* throw something
+* do nothing
+* proceed to invoke the real task
+* simulate an interruption
+* notify a condition
+* wait for a condition
+* wait for the execution to be cancelled by the code under test via the returned future
+
+Some of these actions can be further customized to be conditional, to repeat for a certain number of attempts, to repeat for multiple attempts until a certain condition occurs or until the execution is cancelled, or again can be delayed. Some actions will terminate an attempt like those that returns or throw something and others will just be waiting for something before moving on to the next action before responding to an attempt.
+
+| Action                          | Description | Cardinality (invoked for how many attempts)| Terminates the attempt (doesn't call the production tasks) | Short-circuiting |
```

Comment:
I know it doesn't look good here but it does render as a nice table

Reply:
I only matched the first column. The other columns are not marked  just because they are variable in length. There is no easy way to do this as it seem it has to be defined on one line 😞